WikiDrips/WikiTableViewController.swift

```diff
@@ -8,21 +8,57 @@
 
 import UIKit
 
-class WikiTableViewController: UITableViewController {
+class WikiTableViewController: UITableViewController, UISearchBarDelegate {
+    
+    // MARK: - View Controller Lifecycle
 
-    var wikiDocs = [[WikiDoc]]()
     let dateFormatter = DateFormatter()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
+        searchBar?.delegate = self
+        
         dateFormatter.dateStyle = .long
         dateFormatter.timeStyle = .short
         
+        tableView.estimatedRowHeight = tableView.rowHeight
+        tableView.rowHeight = UITableViewAutomaticDimension
+    }
+    
+    // MARK: - Searching
+    
+    var wikiDocs = [[WikiDoc]]()
+    
+    var searchText: String? = "Salesforce" {
+        didSet {
+            searchBar?.text = searchText
+            wikiDocs.removeAll()
+            tableView.reloadData() // clear out the table view
+            search()
+        }
+    }
+    
+    @IBOutlet weak var searchBar: UISearchBar? {
```

Comment:
Before you get too far on search logic you may want to look into UISearchController